rocks_common.ml

```diff
@@ -84,8 +84,10 @@ let make_destroy t destructor =
       destructor
       (t @-> returning void) in
   fun t ->
-  inner t;
-  t.valid <- false
+  if t.valid then begin
```

Comment:
I'm still not sure if we want to be silent about the caller destroying a resource twice ... it indicates a bug that should be handled.
Do you still need this change?

Reply:
I'm not sure should this be considered as a bug. Anyway, throwing an exception from finalyzer is a bad idea.
I have no code that calls finalyzer twice.